chalice/deploy/deployer.py

```diff
@@ -606,6 +606,7 @@ def _build_lambda_function(self,
             security_group_ids=security_group_ids,
             subnet_ids=subnet_ids,
             reserved_concurrency=config.reserved_concurrency,
+            layers=config.layers
```

Comment:
We should add the same logic we have for subnet/security groups where we default to an empty list if no configuration is provided in `.chalice/config.json`.  One of the things I noticed while trying out this PR was that if I had a layer in my chalice config, deploy my app, then remove the layer from my config, the layer will still be part of the function configuration.

Reply:
@jamesls Thanks for your feedback!
We are using AWS layers to give the possibility to use dependencies of operating system unavailable in lambda functions by default.
For instance, we need to use git and ssh clients which is not installed in AMI.
We install git and ssh from source files and build it statically, after that we compressed
installed data and store it as lambda layers.
Some python libraries require dependencies that need to be installed in OS.
For example, GitPython requires executable git binary.